integration-tests/slack/slack-utils.js

```diff
@@ -0,0 +1,148 @@
+    return filtered.map(item => item.url)
+  } catch (e) {
+    console.error(`**  ERROR: Encountered unexpected error when getting CircleCI build_number: ${buildNum} artifacts:`, e)
+    throw e
+  }
+}
+
+const getFailedTestFileName = aggregatedResult => {
+  const failedTests = []
+  _.forEach(testResult => {
+    if (testResult.numFailingTests > 0) {
+      failedTests.push(parse(testResult.testFilePath).name)
+    }
+  }, aggregatedResult.testResults)
+  return failedTests
+}
+
+/**
+ *
+ * @returns {Promise<Array>}
+ */
+const getFailedTests = async () => {
+  const headers = { Accept: 'application/json' }
+  const urls = await fetchCircleJobArtifacts()
+  const tests = []
+  for (const url of urls) {
+    try {
+      // Parse all tests-summary JSON for failed tests
+      const response = await fetch(url, { method: 'GET', headers })
+      const json = await response.json()
+      const failedTest = getFailedTestFileName(json)
+      tests.push(...failedTest)
+    } catch (e) {
+      console.error(`**  ERROR: Encountered unexpected error when getting CircleCI artifacts tests-summary file: ${url}.`, e)
+      throw e
+    }
+  }
+  return tests
+}
+
+/**
+ *
+ * @param { Array[string] } failedTests
+ * @returns { Map<string, Array[string]> }
+ */
+const getChannelsNotifyFailed = failedTests => {
+  const channels = new Map()
+  const data = JSON.parse(fs.readFileSync('./slack/slack-notify-channels.json', 'utf8'))
+  const { fail: jsonItems } = data
+
+  _.forEach(item => {
+    const id = _.get(['id'], item)
+    const { tests } = item
+    const names = _.map(item => item.name, tests)
+    channels.set(id, names)
+  }, jsonItems)
+
+  // Construct a map object: key: channel_id, value: test_names array
+  const idsMap = new Map()
+  _.forEach(test => {
+    const filtered = Array.from(channels.keys())
+      .map(key => channels.get(key).includes(test) || channels.get(key).length === 0 ? key : undefined)
+      .filter(key => key)
+    if (filtered.length === 0) {
+      throw new Error(`Test: ${test} was not found in slack-notify-channels.json`)
+    }
+    _.forEach(k => idsMap.has(k) ? idsMap.get(k).push(test) : idsMap.set(k, new Array(test)), filtered)
+  }, failedTests)
+
+  return idsMap
+}
+
+/**
+ * Slack to notify job failed. Message contains failed test names.
+ * @param { Array[string] } failedTests
+ * @returns { Promise<void> }
+ */
+// eslint-disable-next-line
+const notifyFailure = async failedTests => {
+  const data = getChannelsNotifyFailed(failedTests)
+  // Slack issue: No way to post the same message to multiple channels at once. https://github.com/slackapi/bolt-js/issues/696
+  _.forEach(async ([k, v]) => {
+    // k is channel_id. v is test_name array.
+    const blocks = printMessageTemplate(1)
+    blocks.push({
+      type: 'section',
+      fields:
+        [
+          {
+            type: 'mrkdwn',
+            text: `:errrrr:  Tests:\n\`\`\`*  ${v.join('\n*  ')}\`\`\``
```

Comment:
👍🏻
There is no Markdown list in Slack. What I have is what Slack recommends to do to make a list. I'll remove header and emoij.

Reply:
There should be a way to render this as something like:
![Screen Shot 2022-08-05 at 1 16 41 PM](https://user-images.githubusercontent.com/1156625/183128272-eaa7c72a-3c70-485f-a8af-cdec039974bd.png)

Instead of:
![Screen Shot 2022-08-05 at 1 17 09 PM](https://user-images.githubusercontent.com/1156625/183128281-2be730db-a483-458f-9aef-85929b4e91b1.png)

I think removing the backticks for the code block formatting would do it.
https://api.slack.com/reference/surfaces/formatting#block-formatting

Reply:
Done. Originally, I wanted to make more contrast.